### PR TITLE
Fix tilemap live editing while game is running

### LINKED ISSUE #77328: Tilemaps no longer reflect changes via live updates
### Godot version

v4.0.3.stable.mono.official [5222a99f5]

### System information

Windows 10

### Issue description

When updating a Tilemap's cells via the editor, the changes are not reflected in the game. This worked fine in 3.5

### Steps to reproduce

1. Create a TileMap and Tileset with a tile you'd like to paint with. (I've been testing with the godot icon.)
2. Make sure "Synchronize Scene Changes" is turned on.
3. Launch your game, and paint some tiles in the TileMap.
4. Save if you want, but, it shouldn't matter. Godot 3.5 behavior didn't require a save for changes to appear.

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by pancelor**
fixes #77328

thanks to @torcado194 for figuring out how to fix this!

I've been using this for a week or two with no problems in my simple use cases (single-layer single-source tilemaps)

### CODE REVIEW

**Comment by groud** [comment]
Hi! Thanks for trying to the solve the issue but this is not the right way to solve the problem.

The `commit_action(false)` call aims to avoid calling the functions on the locally edited TileMap for performance reasons. The argument cannot be removed like that as it would impact editing performances.

The issue must be solved in UndoRedo. I think solving the issue should be done by still calling the callbacks when `p_execute` is false. This means:
- Updating `UndoRedo::commit_action' to always call `_process_operation_list ` but with and added `p_execute` argument.
- Adding a `p_execute` argument to `_process_operation_list` 
- Only execute code when `p_execute` is true (the `callable.callp(nullptr, 0, ret, ce)` part), but still notify the debugger via the callback (check how `method_callback` is used)

**Comment by pancelor** [comment]
Ahh that makes more sense; thank you for the detailed steps! lmk how it looks

File: core/object/undo_redo.cpp [hunk 1]
```diff
@@ -337,18 +335,20 @@ void UndoRedo::_process_operation_list(List<Operation>::Element *E) {
 
 		switch (op.type) {
 			case Operation::TYPE_METHOD: {
```
Comment (KoBeWi) [inline core/object/undo_redo.cpp:337]: What about `TYPE_PROPERTY`?

Comment (pancelor) [reply core/object/undo_redo.cpp:337]: Oh! I overlooked that -- updated now

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by caimantilla** [comment]
Did this break again?
I've been struggling figuring out how to make my own `Node` type work with live editing, which led me here.
So, I added a `TileMap`, and the tiles I paint only show up in-game once I restart...

**Comment by KoBeWi** [comment]
Can't reproduce the problem, the TileMap updates normally for me.

**Pull request merged**